frontend/pipeline.yml

```
@@ -6,6 +6,11 @@ variables:
   DOCS_VERSION: development
   MENDER_IMAGE_GUI: ${MENDER_IMAGE_REGISTRY}/${MENDER_IMAGE_REPOSITORY}/gui:${MENDER_IMAGE_TAG}
   FF_TIMESTAMPS: true
+  FRONTEND_DIR: ${CI_PROJECT_DIR}/frontend
+  LOGS_DIR: ${CI_PROJECT_DIR}/logs
+  COVERAGE_DIR: ${FRONTEND_DIR}/coverage
+  SCREENSHOTS_DIR: ${FRONTEND_DIR}/screenshots
+  JUNIT_DIR: ${FRONTEND_DIR}
```

How about instead of defining 5 different variables that points to frontend artifacts, consolidate everything into a `FRONTEND_ARTIFACTS_DIR` pointing to the root of all artifacts and update the jobs the scripts that generate these artifacts to write to sub-directories inside this path?
In that case it doesn't matter what you define the variable as and there will be less boilerplate dealing with the correct environment variables.

```suggestion
  FRONTEND_ARTIFACTS_ROOT: "${CI_PROJECT_DIR:-$(git rev-parse --show-toplevel)}/frontend/artifacts"
```

---

I like this approach!

It does rely on git, and I am not sure how it's going to work for things like shallow clones, but I also cannot see the $CI_PROJECT_DIR to not have a value. Something must've gone very wrong if that happened. Either way, a cool fallback.

---

I looked into the artifacts root path changes and it won't be as straightforward. 

I'd need to add something akin to `path.join(frontendDir, 'artifacts');` into `vitest.config.ts` and `playwright.config.ts` because they are set to output to their root which would be `FRONTEND_DIR` and not `FRONTEND_DIR/artifacts` 

I'd rather not change it this way, unless we really really want it.

I can go back to a partial approach. Keep the ` FRONTEND_DIR` variable
```yaml
variables:
  FRONTEND_DIR: ${CI_PROJECT_DIR:-$(git rev-parse --show-toplevel)}/frontend
```
but don't do anything extra for the different artifacts dirs. And for the job artifacts just do:
```yaml
artifacts:
    paths:
      - ${FRONTEND_DIR}/coverage
      - ${FRONTEND_DIR}/screenshots
      - ${FRONTEND_DIR}/junit
```

What do you think?

---

Actually turns out that
```yaml
FRONTEND_DIR: ${CI_PROJECT_DIR:-$(git rev-parse --show-toplevel)}/frontend
```
won't work because gitlab will not do command substitution, so I changed it to simply 
```yaml
FRONTEND_DIR: ${CI_PROJECT_DIR}/frontend
```

---

@alfrunes What's your take on he changes now?